python_modules/dagster/dagster/_core/telemetry.py

```diff
@@ -523,6 +528,40 @@ def log_action(
         )
 
 
+def log_step_event(event: DagsterEvent, pipeline_context: PlanOrchestrationContext):
+    if event.is_step_start:
+        log_action(
+            instance=pipeline_context.instance,
+            action=STEP_START_EVENT,
+            client_time=datetime.datetime.now(),
+            metadata={
+                "run_id_hash": hash_name(pipeline_context.run_id),
+                "step_key_hash": hash_name(event.step_key),
+            },
+        )
```

Comment:
nit: would probably be more terse to 

1/ have an early return for non-step events
2/ use the conditionals to determine the `action` and `metadata`, and have one `log_action` invocation once those are determined

```suggestion
def log_step_event(event: DagsterEvent, pipeline_context: PlanOrchestrationContext):
    if not any((event.is_step_start, event.is_step_success, event.is_step_failure)):
        return
    
    metadata = {
        "run_id_hash": hash_name(pipeline_context.run_id),
        "step_key_hash": hash_name(event.step_key),
    }
                
    if event.is_step_start:
        action = STEP_START_EVENT
```